src/domino.py

```diff
@@ -0,0 +1,208 @@
+        volumes.append(bind_path)
+
+        # Use its --output_folder argument to set the output file prefix to specify an absolute path and prefix
+        out_dir = Path(output_file).parent
+        out_dir.mkdir(parents=True, exist_ok=True)
+        bind_path, mapped_out_dir = prepare_volume(str(out_dir), work_dir)
+        volumes.append(bind_path)
+
+        ########
+
+        bind_path, mapped_slices_dir = prepare_volume('slices.txt', work_dir)
+        volumes.append(bind_path)
+        # /spras/ADFJGFD/slices.txt
+
+        slicer_command = ['slicer',
+            '--network_file', network_file,
+            '--output_file', mapped_slices_dir]
+
+        print('Running slicer with arguments: {}'.format(' '.join(slicer_command)), flush=True)
+
+        container_framework = 'singularity' if singularity else 'docker'
+        slicer_out = run_container(container_framework,
+                            'otjohnson/domino',
+                            slicer_command,
+                            volumes,
+                            work_dir)
+        print(slicer_out)
+
+        ########
+
+        # Make the Python command to run within the container
+        command = ['domino',
+                   '--active_genes_files', node_file,
+                   '--network_file', network_file,
+                   '--slices_file', mapped_slices_dir,
+                   '--output_folder', mapped_out_dir,
+                   '--parallelization', '1',
+                   '--visualization', 'true']
+
+        # Add optional arguments
+        if use_cache is not True:
+            command.extend(['-c', 'false'])
+        if slices_threshold is not None:
+            command.extend(['-sth', str(slices_threshold)])
+        if module_threshold is not None:
+            command.extend(['-mth', str(module_threshold)])
+
+        print('Running DOMINO with arguments: {}'.format(' '.join(command)), flush=True)
+
+        # container_framework = 'singularity' if singularity else 'docker'
+        domino_out = run_container(container_framework,
+                            'otjohnson/domino',
+                            command,
+                            volumes,
+                            work_dir)
+        print(domino_out)
+
+        ########
+
+        # Path(mapped_slices_dir).unlink(missing_ok=True)
+        # Path(out_dir, 'modules.out').unlink(missing_ok=True)
+        #for domino_output in out_dir.glob('modules.out'):
+        #    domino_output.unlink(missing_ok=True)
+
+        # domino creates a new folder in out_dir to output its modules files into /active_genes
+        out_modules_dir = Path(out_dir, 'active_genes')
+
+        # concatenate each module html file into one big file
+        with open(output_file, "w") as fo:
+            for html_file in out_modules_dir.glob('module_*.html'):
+                with open(html_file,'r') as fi:
+                    fo.write(fi.read())
+                # Path(html_file).unlink(missing_ok=True)
```

Comment:
I'm not sure whether we want to delete these or leave them. They are potentially useful visualizations.

Reply:
I decided we should leave them.